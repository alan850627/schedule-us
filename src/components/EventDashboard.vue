<template>
  <div class="event-dashboard">
    <b-container v-if="validEvent">
      <b-row>
        <b-col sm="8">
          <b-card>
            <time-table
              :id="event.timeTableId"
              :editable="false"
              :username="viewUsername"
              :view-single="viewUsername.length > 0">
            </time-table>
          </b-card>
        </b-col>
        <b-col sm="4">
          <b-button :href="`/#/edit-calendar/${eventId}`">Edit Availability</b-button>
          <b-button @click="emailModal=true">Email Everyone</b-button>
          <b-card class="my-4" >
            <h2> {{event.name}} </h2>
            <p> {{event.description}} </p>
          </b-card>
          <b-card title="Responses" class="my-4">
            <span v-if="Object.keys(event.response).length > 0">
              <div v-for="(res, name) in event.response" v-bind:key="name"
                v-on:mouseover="updateView(name)" v-on:mouseleave="updateView('')">
                {{ name }}
              </div>
            </span>
            <span v-else>
              none
            </span>
          </b-card>
          <chat-box
            class="my-4"
            :id="event.chatId"
            :username="username">
          </chat-box>
        </b-col>
      </b-row>
    </b-container>
    <b-modal v-model="emailModal" id="emailModal" size="lg" title="Email Everyone">
      <p>Warning: Emails only sent to those who opt in and provided emails in their responses.</p>
      <b-input-group prepend="Your Email">
        <b-form-input v-model="replyToEmail" type="email" placeholder="Enter your email" required></b-form-input>
      </b-input-group>
      <b-input-group prepend="Subject">
        <b-form-input v-model="emailSubject" type="text" placeholder="Enter your email subject here" required></b-form-input>
      </b-input-group>
      <b-input-group prepend="Additional Recipients">
        <b-form-input v-model="additionalEmails" type="text" placeholder="Separate emails by commas"></b-form-input>
      </b-input-group>
      <b-input-group prepend="Email content">
        <b-form-textarea id="textarea1" v-model="emailContent" placeholder="We've decided to schedule our event at..." :rows="8" required></b-form-textarea>
      </b-input-group>
      <div slot="modal-footer" class="w-100">
        <b-btn class="float-right px-3 ml-2" variant="primary" @click="sendEmail">
          Send
        </b-btn>
        <b-btn class="float-right" @click="emailModal=false">
          Cancel
        </b-btn>
      </div>
    </b-modal>
  </div>
</template>

<script>
import db from '../firebaseInit'
import TimeTable from './calendar/TimeTable'
import ChatBox from './chat/ChatBox'

export default {
  name: 'EventDashboard',

  components: {
    TimeTable,
    ChatBox
  },

  props: {
    username: {
      type: String,
      default: ''
    }
  },

  computed: {
    validEvent: function () {
      return Object.keys(this.event).length > 0
    }
  },

  data () {
    return {
      eventId: '',
      viewUsername: '',
      emailSubject: '',
      emailContent: '',
      replyToEmail: '',
      emailModal: false,
      additionalEmails: '',
      event: {}
    }
  },

  methods: {
    updateView: function (name) {
      this.viewUsername = name
    },
    sendEmail: function () {
      // let emailRef = db.collection('emails').doc()
      // let resp = this.additionalEmails
      // Object.keys(this.event.response).forEach((user) => {
      //   resp += `,${this.event.response[user].email}`
      // })
      // let emailObj = {
      //   bcc: resp,
      //   subject: this.emailSubject,
      //   text: this.emailContent,
      //   replyTo: this.replyToEmail,
      //   eventId: this.eventId
      // }
      // emailRef.set(emailObj).then(() => {
      //   this.emailSubject = ''
      //   this.emailContent = ''
      //   this.replyToEmail = ''
      //   this.emailModal = false
      // }).catch((error) => {
      //   alert('Problem with server... Try again. \nError: ' + error)
      // })
    }
  },

  watch: {
    // '$route' (to, from) {
    //   react to route changes...
    //   this.eventId = to.params.eventId
    //   this.$bind('event', db.collection('events').doc(this.eventId))
    // }
  },

  mounted () {
    // this.eventId = this.$route.params.eventId
    // this.$bind('event', db.collection('events').doc(this.eventId))
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
</style>
