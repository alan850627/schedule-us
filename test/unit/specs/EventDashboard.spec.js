import EventDashboard from '@/components/EventDashboard'
import { shallow } from 'vue-test-utils'

describe('EventDashboard.vue', () => {
  it('has the expected html structure', () => {
    let cmp = shallow(EventDashboard, {
      propsData: {
        username: 'test-user'
      }
    })
    cmp.vm.eventId = 'mL5TyA3eVWObYeUzh5dS'
    expect(cmp.element).toMatchSnapshot()
    expect(cmp.vm).toBeDefined()
  })
  it('change view', () => {
    let cmp = shallow(EventDashboard, {
      propsData: {
        username: 'test-user'
      }
    })
    cmp.vm.eventId = 'mL5TyA3eVWObYeUzh5dS'
    cmp.vm.updateView('test-user')
    expect(cmp.vm.viewUsername).toBe('test-user')
  })
})
