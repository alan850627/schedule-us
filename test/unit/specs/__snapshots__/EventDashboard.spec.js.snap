// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDashboard.vue has the expected html structure 1`] = `
<div
  class="event-dashboard"
>
  <!---->
   
  <b-modal
    id="emailModal"
    size="lg"
    title="Email Everyone"
  >
    <p>
      Warning: Emails only sent to those who opt in and provided emails in their responses.
    </p>
     
    <b-input-group
      prepend="Your Email"
    >
      <b-form-input
        placeholder="Enter your email"
        required=""
        type="email"
      />
    </b-input-group>
     
    <b-input-group
      prepend="Subject"
    >
      <b-form-input
        placeholder="Enter your email subject here"
        required=""
        type="text"
      />
    </b-input-group>
     
    <b-input-group
      prepend="Additional Recipients"
    >
      <b-form-input
        placeholder="Separate emails by commas"
        type="text"
      />
    </b-input-group>
     
    <b-input-group
      prepend="Email content"
    >
      <b-form-textarea
        id="textarea1"
        placeholder="We've decided to schedule our event at..."
        required=""
        rows="8"
      />
    </b-input-group>
     
    <div
      class="w-100"
      slot="modal-footer"
    >
      <b-btn
        class="float-right px-3 ml-2"
        variant="primary"
      >
        
        Send
      
      </b-btn>
       
      <b-btn
        class="float-right"
      >
        
        Cancel
      
      </b-btn>
    </div>
  </b-modal>
</div>
`;
